src/main/java/wsu/edu/SnakePane.java

```diff
@@ -0,0 +1,117 @@
+package wsu.edu;
+
+import java.util.LinkedList;
+import java.util.List;
+import javafx.animation.KeyFrame;
+import javafx.animation.Timeline;
+import javafx.scene.Scene;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.Pane;
+import javafx.scene.paint.Color;
+import javafx.scene.shape.Circle;
+import javafx.stage.Stage;
+import javafx.util.Duration;
+
+
+public class SnakePane {
+  private List<Circle> createSnake() {
+    List<Circle> snake = new LinkedList<>();
+    Circle head = new Circle(300, 200, 10, Color.RED);
+    snake.add(head);
+
+    int numCircles = 10;
+    for (int i = 1; i <= numCircles; i++){
+      Circle bodySecment = new Circle(300 - i * 20, 200, 10, Color.LIMEGREEN);
+      snake.add(bodySecment);
+    }
+    return snake;
+  }
+  private void drawSnake(List<Circle> snake, Pane pane ){
+    pane.getChildren().addAll(snake);
+  }
+  private void moveSnakeRight(List<Circle> snake){
+    Circle oldHead = snake.get(0);
+    Circle newHead = snake.remove(snake.size() - 1);
+    oldHead.setFill(Color.LIMEGREEN);
+    newHead.setFill(Color.RED);
+    newHead.setCenterX((oldHead.getCenterX() + 20) % 600);
+    newHead.setCenterY(oldHead.getCenterY());
+    snake.add(0, newHead);
+  }
+  private void moveSnakeLeft(List<Circle> snake){
+    Circle oldHead = snake.get(0);
+    Circle newHead = snake.remove(snake.size() - 1);
+    oldHead.setFill(Color.LIMEGREEN);
+    newHead.setFill(Color.RED);
+    newHead.setCenterX((oldHead.getCenterX() - 20 + 600) % 600);
+    newHead.setCenterY(oldHead.getCenterY());
+    snake.add(0, newHead);
+  }
+  private void moveSnakeDown(List<Circle> snake){
+    Circle oldHead = snake.get(0);
+    Circle newHead = snake.remove(snake.size() - 1);
+    oldHead.setFill(Color.LIMEGREEN);
+    newHead.setFill(Color.RED);
+    newHead.setCenterY((oldHead.getCenterY() + 20) % 400);
+    newHead.setCenterX(oldHead.getCenterX());
+    snake.add(0, newHead);
+  }
+  private void moveSnakeUp(List<Circle> snake){
+    Circle oldHead = snake.get(0);
+    Circle newHead = snake.remove(snake.size() - 1);
+    oldHead.setFill(Color.LIMEGREEN);
+    newHead.setFill(Color.RED);
+    newHead.setCenterY((oldHead.getCenterY() - 20 + 400) % 400);
+    newHead.setCenterX(oldHead.getCenterX());
+    snake.add(0, newHead);
+  }
+
+
+  public void start(Stage stage) {
+    AnchorPane anchorPane = new AnchorPane();
+    anchorPane.setPrefSize(600,400);
+
+    List<Circle> snake = createSnake();
+    drawSnake(snake, anchorPane);
+
+    Timeline timeline = new Timeline();
+    timeline.setCycleCount(Timeline.INDEFINITE);
+    timeline.getKeyFrames().add(new KeyFrame(Duration.millis(100), event1 -> moveSnakeRight(snake)));
+    timeline.play();
+
+    Scene scene = new Scene(anchorPane);
+    stage.setScene(scene);
+    scene.setOnKeyPressed(event -> {
+      switch (event.getCode()){
+        case UP:
+          timeline.stop();
```

Comment:
Yes, this will work. You often come up with a solution that is completely different than other students, Zach. 

An alternate way to look at this is by viewing the Timeline animation as a "motor" that move the snake while the keyboard event steers the snake -- in this implementation, you have a variable that specifies the direction of the snake movement. The keyboard event changes the value of the variable. The Timeline event moves the snake according to the variable's value.